yb-voyager/src/migassessment/gather.go

```diff
@@ -462,11 +469,9 @@ func runGatherAssessmentMetadataScriptBuffered(
 	}
 
 	// Report starting status
-	if progressChan != nil {
-		progressChan <- NodeProgress{
-			NodeName: nodeName,
-			Stage:    "Starting collection...",
-		}
+	progressChan <- NodeProgress{
```

Comment:
since you removed the nil checks everywhere, you should do one nil check at the beginning of the function and return err.